src/test/java/com/ernie/app/ticket/TicketServiceTest.java

```diff
@@ -1,132 +1,251 @@
-//            for (String s : tickets) {
-//                System.out.println(s);
-//            }
-//        } catch (
-//                IOException e) {
-//            e.printStackTrace();
-//        }
-    }
-}
+
+    @Mock
+    private TicketRepository ticketRepository;
+
+    @Mock
+    private UserRepository userRepository;
+
+    @InjectMocks
+    private TicketServiceImpl ticketService;
+
+    @Value("${user_update_and_ticket_creation.enabled}")
+    public boolean isUserUpdateAndTicketCreationEnabled;
+
+    private Ticket ticket;
+    private User user;
+    private UUID ticketId;
+
+    @BeforeEach
+    void setUp() {
+        ticketId = UUID.randomUUID();
+        ticket = new Ticket();
+        ticket.setId(ticketId);
+        ticket.setPrice(20);
+
+        user = new User("TestUserName");
+        user.setId(UUID.randomUUID());
+
+        isUserUpdateAndTicketCreationEnabled = true;
+    }
+
+    @Test
+    void saveTicket_positiveTest() {
+        when(ticketRepository.save(ticket)).thenReturn(ticket);
+
+        Ticket savedTicket = ticketService.saveTicket(ticket);
+
+        assertNotNull(savedTicket);
+        assertEquals(ticketId, savedTicket.getId());
+        verify(ticketRepository, times(1)).save(ticket);
+    }
+    @Test
+    void saveTicket_negativeTest_creationDisabled() {
+        ticketService.isUserUpdateAndTicketCreationEnabled = false;
+
+        assertThrows(IllegalArgumentException.class, () -> ticketService.saveTicket(ticket));
+        verify(ticketRepository, never()).save(ticket);
+    }
+    @Test
+    void saveTicket_cornerCase_nullTicket() {
+        assertThrows(NullPointerException.class, () -> ticketService.saveTicket(null));
+        verify(ticketRepository, never()).save(any());
+    }
+
+
+    @Test
+    void getTicketById_positiveTest() {
+        when(ticketRepository.getReferenceById(ticketId)).thenReturn(ticket);
+
+        Ticket foundTicket = ticketService.getTicketById(ticketId);
+
+        assertNotNull(foundTicket);
+        assertEquals(ticketId, foundTicket.getId());
+        verify(ticketRepository, times(1)).getReferenceById(ticketId);
+    }
+    @Test
+    void getTicketById_negativeTest_nonExistentId() {
+        UUID nonExistentId = UUID.randomUUID();
+        when(ticketRepository.getReferenceById(nonExistentId)).thenReturn(null);
+
+        Ticket result = ticketService.getTicketById(nonExistentId);
+
+        assertNull(result);
+        verify(ticketRepository, times(1)).getReferenceById(nonExistentId);
+    }
+    @Test
+    void getTicketById_cornerCase_nullId() {
+        assertThrows(IllegalArgumentException.class, () -> ticketService.getTicketById(null));
+        verify(ticketRepository, never()).getReferenceById(any());
+    }
+
+
+    @Test
+    void getAllTickets_positiveTest() {
+        when(ticketRepository.findAll()).thenReturn(List.of(ticket));
+
+        List<Ticket> tickets = ticketService.getAllTickets();
```

Comment:
evry time when you return something you need to compare this result with some expected result